packages/playground-server/src/middleware/matchmake.ts

```diff
@@ -0,0 +1,59 @@
+import { Context } from "koa";
+import "koa-body"; // See: https://github.com/dlau/koa-body/issues/109
+
+import { createErrorResponse, createErrorResponseForDatabase } from "../api";
+import { matchmakeUser } from "../db";
+import {
+  ApiResponse,
+  ErrorCode,
+  MatchmakeRequest,
+  MatchmakeResponseData,
+  PlaygroundUserData
+} from "../types";
+
+function validateRequest(params: MatchmakeRequest): ApiResponse {
+  if (!params.userAddress) {
+    return createErrorResponse(400, ErrorCode.UserAddressRequired);
+  }
+
+  return { ok: true };
+}
+
+export default function matchmake() {
+  return async (ctx: Context, next: () => Promise<void>) => {
+    // TODO: Do we need signature validation here?
```

Comment:
You need to be logged in to call `matchmake`. Let's add this when we add the login route.